app/models/user.rb

```diff
@@ -183,7 +183,10 @@ def direct_filters
 
   def expire_caches
     if login_changed?
-      self.works.each{ |work| work.touch }
+      self.works.each do |work| 
+        work.touch
+        work.expire_caches
```

Comment:
It seems to me that work.touch should expire its caches--I'd expect it to. Can you make that happen rather than having this manually call the method as well as touching the work?